### PR TITLE
string-functions: Explain binary-as-hex behavior

### PR DESCRIPTION
**Pull request by hfxsd**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [x] v8.3 (TiDB 8.3 versions)
- [x] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs/pull/19117
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by lcwangchao — APPROVED** [review]

**Review by lcwangchao — APPROVED** [review]

File: functions-and-operators/string-functions.md [hunk 1]
```diff
@@ -634,7 +634,11 @@ mysql> SELECT FROM_BASE64('MTIzNDU2');
 - 如果输入参数为数字，`HEX(n)` 返回 `n` 的十六进制字符串表示。该函数将参数 `n` 视为 `BIGINT` 数字，相当于 `CONV(n, 10, 16)`。
 - 如果输入参数为 `NULL`，该函数返回 `NULL`。
 
-示例：
+> **注意：**
+>
+> 在 MySQL 客户端中，[`--binary-as-hex`](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_binary-as-hex) 选项在交互模式下默认启用，这会导致客户端将无法识别字符集的数据显示为[十六进制字面值 (Hexadecimal literal)`](https://dev.mysql.com/doc/refman/8.0/en/hexadecimal-literals.html)。你可以使用 `--skip-binary-as-hex` 选项来禁用此行为。
```
Comment (hfxsd) [inline functions-and-operators/string-functions.md:639]:
```suggestion
> 在 MySQL 客户端中，[`--binary-as-hex`](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_binary-as-hex) 选项在交互模式下默认启用，这会导致客户端将无法识别字符集的数据显示为[十六进制字面量 (Hexadecimal literal)`](https://dev.mysql.com/doc/refman/8.0/en/hexadecimal-literals.html)。你可以使用 `--skip-binary-as-hex` 选项来禁用此行为。
```

File: functions-and-operators/string-functions.md [hunk 2]
```diff
@@ -2220,7 +2224,8 @@ SELECT UCASE('bigdata') AS result_upper, UCASE(null) AS result_null;
 
 > **注意：**
 >
-> 传入的字符串必须是合法的十六进制数值，包含 `0~9`、`A~F`、`a~f`，如果为 `NULL` 或超出该范围，则返回 `NULL`。
+> - 传入的字符串必须是合法的十六进制数值，包含 `0~9`、`A~F`、`a~f`，如果为 `NULL` 或超出该范围，则返回 `NULL`。
+> - 在 MySQL 客户端中，[`--binary-as-hex`](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_binary-as-hex) 选项在交互模式下默认启用，这会导致客户端将无法识别字符集的数据显示为[十六进制字面值 (Hexadecimal literal)`](https://dev.mysql.com/doc/refman/8.0/en/hexadecimal-literals.html)。你可以使用 `--skip-binary-as-hex` 选项来禁用此行为。
```
Comment (hfxsd) [inline functions-and-operators/string-functions.md:2228]:
```suggestion
> - 在 MySQL 客户端中，[`--binary-as-hex`](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_binary-as-hex) 选项在交互模式下默认启用，这会导致客户端将无法识别字符集的数据显示为[十六进制字面量 (Hexadecimal literal)`](https://dev.mysql.com/doc/refman/8.0/en/hexadecimal-literals.html)。你可以使用 `--skip-binary-as-hex` 选项来禁用此行为。
```

File: functions-and-operators/string-functions.md [hunk 3]
```diff
@@ -634,7 +634,11 @@ mysql> SELECT FROM_BASE64('MTIzNDU2');
 - 如果输入参数为数字，`HEX(n)` 返回 `n` 的十六进制字符串表示。该函数将参数 `n` 视为 `BIGINT` 数字，相当于 `CONV(n, 10, 16)`。
 - 如果输入参数为 `NULL`，该函数返回 `NULL`。
 
-示例：
+> **注意：**
+>
+> 在 MySQL 客户端中，[`--binary-as-hex`](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_binary-as-hex) 选项在交互模式下默认启用，这会导致客户端将无法识别字符集的数据显示为[十六进制字面量 (Hexadecimal literal)](https://dev.mysql.com/doc/refman/8.0/en/hexadecimal-literals.html)。你可以使用 `--skip-binary-as-hex` 选项来禁用此行为。
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:639]:
```suggestion
> 在 MySQL 客户端中，[`--binary-as-hex`](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_binary-as-hex) 选项在交互模式下默认启用，这会导致客户端将无法识别的字符集数据显示为[十六进制字面量 (Hexadecimal literal)](https://dev.mysql.com/doc/refman/8.0/en/hexadecimal-literals.html)。你可以使用 `--skip-binary-as-hex` 选项来禁用此行为。
```

File: functions-and-operators/string-functions.md [hunk 4]
```diff
@@ -2220,7 +2224,8 @@ SELECT UCASE('bigdata') AS result_upper, UCASE(null) AS result_null;
 
 > **注意：**
 >
-> 传入的字符串必须是合法的十六进制数值，包含 `0~9`、`A~F`、`a~f`，如果为 `NULL` 或超出该范围，则返回 `NULL`。
+> - 传入的字符串必须是合法的十六进制数值，包含 `0~9`、`A~F`、`a~f`，如果为 `NULL` 或超出该范围，则返回 `NULL`。
+> - 在 MySQL 客户端中，[`--binary-as-hex`](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_binary-as-hex) 选项在交互模式下默认启用，这会导致客户端将无法识别字符集的数据显示为[十六进制字面量 (Hexadecimal literal)](https://dev.mysql.com/doc/refman/8.0/en/hexadecimal-literals.html)。你可以使用 `--skip-binary-as-hex` 选项来禁用此行为。
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:2228]:
```suggestion
> - 在 MySQL 客户端中，[`--binary-as-hex`](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_binary-as-hex) 选项在交互模式下默认启用，这会导致客户端将无法识别的字符集数据显示为[十六进制字面量 (Hexadecimal literal)](https://dev.mysql.com/doc/refman/8.0/en/hexadecimal-literals.html)。你可以使用 `--skip-binary-as-hex` 选项来禁用此行为。
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**